influxdb3_write/src/write_buffer/mod.rs

```diff
@@ -568,6 +523,88 @@ impl<W: Wal, T: TimeProvider> ChunkContainer for WriteBufferImpl<W, T> {
     }
 }
 
+impl<W: Wal, T: TimeProvider> LastCacheManager for WriteBufferImpl<W, T> {
+    fn last_cache_provider(&self) -> Arc<LastCacheProvider> {
+        Arc::clone(&self.last_cache)
+    }
+
+    /// Create a new last-N-value cache in the specified database and table, along with the given
+    /// parameters.
+    ///
+    /// Returns the name of the newly created cache, or `None` if a cache was not created, but the
+    /// provided parameters match those of an existing cache.
+    #[allow(clippy::too_many_arguments)]
+    fn create_last_cache(
+        &self,
+        db_name: &str,
+        tbl_name: &str,
+        cache_name: Option<&str>,
+        count: Option<usize>,
+        ttl: Option<Duration>,
+        key_columns: Option<Vec<String>>,
+        value_columns: Option<Vec<String>>,
+    ) -> Result<Option<LastCacheDefinition>, Error> {
+        let cache_name = cache_name.map(Into::into);
+        let catalog = self.catalog();
+        let sequence = catalog.sequence_number();
+        let db_schema = catalog.db_schema(db_name).ok_or(Error::DbDoesNotExist)?;
+        let table = db_schema
+            .get_table(tbl_name)
+            .ok_or(Error::TableDoesNotExist)?;
+        let schema = table.schema.clone();
+        if let Some(info) = self.last_cache.create_cache(CreateCacheArguments {
+            db_name: db_name.to_string(),
+            tbl_name: tbl_name.to_string(),
+            schema,
+            cache_name,
+            count,
+            ttl,
+            key_columns,
+            value_columns,
+        })? {
+            let mut db_schema = db_schema.as_ref().clone();
+            let mut table = table.clone();
+            table.add_last_cache(info.clone());
+            assert!(
+                db_schema
+                    .tables
+                    .insert(tbl_name.to_string(), table)
+                    .is_some(),
+                "there should have been an existing table when adding last cache"
+            );
```

Comment:
Since I cloned the DB schema I should be able to just `get_mut` the table and modify it directly, instead of cloning it again.